model/project_ref.go

```diff
@@ -465,15 +459,14 @@ func (p *ProjectRef) MergeWithProjectConfig(version string) error {
 		if projectConfig.TaskAnnotationSettings != nil {
 			pRefToMerge.TaskAnnotationSettings = *projectConfig.TaskAnnotationSettings
 		}
-		if projectConfig.CommitQueue != nil {
-			pRefToMerge.CommitQueue = *projectConfig.CommitQueue
-		}
 		if projectConfig.TaskSync != nil {
 			pRefToMerge.TaskSync = *projectConfig.TaskSync
 		}
 		reflectedRef := reflect.ValueOf(p).Elem()
 		reflectedConfig := reflect.ValueOf(pRefToMerge)
 		recursivelySetUndefinedFields(reflectedRef, reflectedConfig)
+		p.PerfEnabled = utility.ToBoolPtr(utility.FromBoolPtr(projectConfig.PerfEnabled) || utility.FromBoolPtr(p.PerfEnabled))
```

Comment:
Why this change? This is actually incorrect bc if PerfEnabled is _supposed_ to be nil (bc both the project config and the project ref are nil) we'll now incorrectly set it to false explicitly.

Reply:
I made this change because I noticed that once a boolean ptr is set, it cannot become nil again, at least with the current way modifying a project ref works on the UI.  In this case if someone toggled PerfEnabled true on the UI, and then later untoggled it, it would be stuck in false rather than nil.  Therefore the project config could never override the project page even though at this point the flag would be "unset" on the project page.  Since having a nil pointer is functionally equivalent to having a false pointer from a code execution perspective since we are using utility.FromBoolPtr I went with this approach.  However open to discussing design alternatives!

Reply:
Hmm I see what you're saying. On the new UI it will be possible to explicitly set the value to false but not yet, and yeah since this is the last step in getting the final merged project ref to be used it would be okay to explicitly set these to true. I'm wondering then if it's maybe just more consistent to also remove these toggles, unless you are attached to them, or else we will need to make modifications to _not_ override an explicitly false value once the new UI is finished.

Reply:
I'm not particularly attached, aliases, periodic builds and build baron configs are definitely the meat here. I'm fine to remove them and perhaps in the future if we get user requests we can add toggles for enabling the performance plugin in version control.